tests/cpp/transforms/scalarize_test.cpp

```diff
@@ -90,49 +141,89 @@ void test_load_scalarize() {
+    TensorType<4 x i32>* %1 = AllocaStmt()
+    TensorType<4 x i32>  %2 = MatrixInitStmt([1, 1, 2, 2])
+    StoreStmt(%1, %2)
   */
   Type *tensor_type = type_factory.get_tensor_type(
       {2, 2}, type_factory.get_primitive_type(PrimitiveTypeID::i32));
-  auto argload_stmt = block->push_back<ArgLoadStmt>(0 /*arg_id*/, tensor_type);
+  Stmt *dest_stmt = block->push_back<AllocaStmt>(tensor_type);
+  dest_stmt->ret_type = type_factory.get_pointer_type(tensor_type);
 
-  std::vector<Stmt *> indices = {};
-  Stmt *src_stmt = block->push_back<ExternalPtrStmt>(
-      argload_stmt, indices);  // fake ExternalPtrStmt
-  src_stmt->ret_type = type_factory.get_pointer_type(tensor_type);
+  auto const_1_stmt = block->push_back<ConstStmt>(TypedConstant(1));
+  auto const_2_stmt = block->push_back<ConstStmt>(TypedConstant(2));
+  std::vector<Stmt *> matrix_init_vals = {const_1_stmt, const_1_stmt,
+                                          const_2_stmt, const_2_stmt};
+  auto matrix_init_stmt =
+      block->push_back<MatrixInitStmt>(std::move(matrix_init_vals));
+  matrix_init_stmt->ret_type = tensor_type;
 
-  block->push_back<T>(src_stmt);
+  block->push_back<LocalStoreStmt>(dest_stmt, matrix_init_stmt);
 
   irpass::scalarize(block.get());
+  irpass::die(block.get());
 
-  EXPECT_EQ(block->size(), 1 /*argload*/ + 1 /*external_ptr*/ + 4 /*const*/ +
-                               4 /*matrix_ptr*/ + 4 /*load*/ +
-                               1 /*matrix_init*/);
+  EXPECT_EQ(block->size(),
+            2 /*const*/ + 1 /*matrix_init*/ + 4 /*alloca*/ + 4 /*store*/);
 
   // Check for scalarized statements
-  EXPECT_EQ(block->statements[2]->is<ConstStmt>(), true);
-  EXPECT_EQ(block->statements[3]->is<MatrixPtrStmt>(), true);
-  EXPECT_EQ(block->statements[4]->is<T>(), true);
+  EXPECT_EQ(block->statements[0]->is<AllocaStmt>(), true);
+  EXPECT_EQ(block->statements[1]->is<AllocaStmt>(), true);
+  EXPECT_EQ(block->statements[2]->is<AllocaStmt>(), true);
+  EXPECT_EQ(block->statements[3]->is<AllocaStmt>(), true);
 
+  EXPECT_EQ(block->statements[4]->is<ConstStmt>(), true);
   EXPECT_EQ(block->statements[5]->is<ConstStmt>(), true);
-  EXPECT_EQ(block->statements[6]->is<MatrixPtrStmt>(), true);
-  EXPECT_EQ(block->statements[7]->is<T>(), true);
+  EXPECT_EQ(block->statements[6]->is<MatrixInitStmt>(), true);
 
-  EXPECT_EQ(block->statements[8]->is<ConstStmt>(), true);
-  EXPECT_EQ(block->statements[9]->is<MatrixPtrStmt>(), true);
-  EXPECT_EQ(block->statements[10]->is<T>(), true);
+  EXPECT_EQ(block->statements[7]->is<LocalStoreStmt>(), true);
+  EXPECT_EQ(block->statements[8]->is<LocalStoreStmt>(), true);
+  EXPECT_EQ(block->statements[9]->is<LocalStoreStmt>(), true);
+  EXPECT_EQ(block->statements[10]->is<LocalStoreStmt>(), true);
+}
 
-  EXPECT_EQ(block->statements[11]->is<ConstStmt>(), true);
-  EXPECT_EQ(block->statements[12]->is<MatrixPtrStmt>(), true);
-  EXPECT_EQ(block->statements[13]->is<T>(), true);
+TEST(Scalarize, ScalarizeLoadAlloca) {
```

Comment:
I didn't quite get the naming - we have `ScalarizeGlobalStore` and `ScalarizeLocalStore` but not `ScalarizeGlobalLoad` and `ScalarizeLocalLoad`?

Reply:
Oh I forgot to test `LocalLoad(AllocaStmt)`. I'll have that fixed. 

In general, `ScalarizedLocalXXX` tests with pointers from `AllocaStmt` and `ScalarizedGlobalXXX` tests with pointers from `ArgLoadStmt`